Add tips for contributing, update State article

netnef opened the pull request:
For the State article, I made these changes:

* Convert the layout to a table (lmk if you don't like it)
* Update the info to match the latest progress
* Add links/issues to some features for reference

Preview: https://netnef.github.io/wiki/state/

sharpenedblade commented:
> There is work in progress support for custom graphics on the Touch Bar.

Its not really WIP anymore, it should just say that its supported

> Trackpad  🟡 Partially working

It should be marked as working, but leave the note there. Same with audio.

AdityaGarg8 commented:
Wi-Fi and Bluetooth drivers are upstreamed. Also, Bluetooth is partially working for BCM4377 devices. On other devices too, it's not very stable.

AdityaGarg8 reviewed (changes requested):
See latest comment

AdityaGarg8 commented:
I added a commit in your pr so that there is no clash.

Redecorating commented on docs/state.md:
```diff
@@ -1,35 +1,26 @@
+|Screen|🟢 Working|🟢 Yes|||
+|USB|🟢 Working|🟢 Yes|||
+|Keyboard|🟢 Working|🔴 No||[apple-bce](https://github.com/t2linux/apple-bce-drv)|
+|Camera|🟢 Working|🟢 Yes|||
```
maybe note that this relies on apple-bce

Redecorating commented on docs/state.md:
```diff
@@ -1,35 +1,26 @@
+|Feature|Status|Upstream|Notes|Issues/links|
+|-|-|-|-|-|
+|Internal Drive / SSD|🟢 Working|🟢 Kernel 5.4||[Filesystem notes](#filesystem-notes)|
+|Screen|🟢 Working|🟢 Yes|||
```
patch to fix overscan on intel gpu is not upstream yet afaik

Redecorating commented on docs/state.md:
```diff
@@ -1,35 +1,26 @@
+|Wi-Fi|🟢 Working|🔴 No|Requires macOS firmware|[Setup guide](https://wiki.t2linux.org/guides/wifi-bluetooth/)|
+|Bluetooth|🟢 Working|🔴 No|Requires macOS firmware only for devices with BCM4377 chip. Also, Bluetooth glitches on devices with BCM4377 Chip if connected to a 2.4 Ghz Wi-Fi connection. Thus, in order to use Bluetooth either turn off your Wi-Fi or use a 5Ghz Wi-Fi connection.|[Setup guide](https://wiki.t2linux.org/guides/wifi-bluetooth/)|
+|Touch Bar|🟢 Working|🔴 No|There is support for the "Touch Bar Keyboard" device configuration, where only the Function Keys or the Media/Brightness Control Keys can be shown. There is work in progress support for custom graphics on the Touch Bar.|[tiny-dfr](https://github.com/kekrby/tiny-dfr)<br>[apple-ib](https://github.com/t2linux/apple-ib-drv)|
+|Suspend|🟡 Partially working|🟢 Yes|A firmware upgrade attached to macOS Sonoma broke suspend. Some users were having difficulty with it even before Sonoma.|[#53](https://github.com/t2linux/T2-Ubuntu-Kernel/issues/53)|
```
can you note that this sonoma issue is in the apple-bce module (because suspend does work if you don't use that module)

Redecorating commented on docs/state.md:
```diff
@@ -1,35 +1,26 @@
+|Suspend|🟡 Partially working|🟢 Yes|A firmware upgrade attached to macOS Sonoma broke suspend. Some users were having difficulty with it even before Sonoma.|[#53](https://github.com/t2linux/T2-Ubuntu-Kernel/issues/53)|
+|Trackpad|🟡 Partially working|🔴 No|Functions, but it is far from the experience on macOS (No force touch or palm rejection). Some models have deadzones on the edges of their trackpads where swipes along the trackpad that start in these deadzones will not be registered.|[apple-bce](https://github.com/t2linux/apple-bce-drv)|
+|Audio|🟡 Partially working|🔴 No|With proper configuration audio can work, however it is not stable in some older kernels and switching between speakers and when using the microphone. Microphone volume is low in some Macs.|[apple-bce](https://github.com/t2linux/apple-bce-drv)|
+|Hybrid Graphics|🟡 Partially working|🟡 Partial|If the device has a dedicated AMD GPU (15 and 16 inch MacBookPro's) as well as an Intel iGPU, the iGPU can be used, but this breaks resume.|[Hybrid Graphics](https://wiki.t2linux.org/guides/hybrid-graphics/)|
```
using igpu no longer breaks resume so you can remove that bit. maybe also mention that toggling the dgpu power doesn't work

Redecorating commented on docs/state.md:
```diff
@@ -1,35 +1,26 @@
+|Keyboard|🟢 Working|🔴 No||[apple-bce](https://github.com/t2linux/apple-bce-drv)|
+|Camera|🟢 Working|🟢 Yes|||
+|Wi-Fi|🟢 Working|🔴 No|Requires macOS firmware|[Setup guide](https://wiki.t2linux.org/guides/wifi-bluetooth/)|
+|Bluetooth|🟢 Working|🔴 No|Requires macOS firmware only for devices with BCM4377 chip. Also, Bluetooth glitches on devices with BCM4377 Chip if connected to a 2.4 Ghz Wi-Fi connection. Thus, in order to use Bluetooth either turn off your Wi-Fi or use a 5Ghz Wi-Fi connection.|[Setup guide](https://wiki.t2linux.org/guides/wifi-bluetooth/)|
```
bluetooth is upstream

Redecorating reviewed (commented):
having a table here is much better

netnef commented:
I updated the State article to address all of the comments here. I also added "Thunderbolt" to the features list.

Preview: https://netnef.github.io/wiki/state/

There's one last thing I wanted to ask about:

> Some models have deadzones on the edges of their trackpads where swipes along the trackpad that start in these deadzones will not be registered.

Is this still an issue? Is there a list of affected models anywhere?

AdityaGarg8 commented:
> I updated the State article to address all of the comments here. I also added "Thunderbolt" to the features list.
> 
> Preview: https://netnef.github.io/wiki/state/
> 
> There's one last thing I wanted to ask about:
> 
> > Some models have deadzones on the edges of their trackpads where swipes along the trackpad that start in these deadzones will not be registered.
> 
> Is this still an issue? Is there a list of affected models anywhere?

Tbh not enough feedback there by the users. Today only mbp 16,2 got fixed.

netnef commented:
Ok, then we can leave the notice there I guess. But I would still leave the status as "working" because for most models it works as well as it possibly can (using libinput, at least). We should probably also add details about how to use the Synaptics driver (with a disclaimer that it's unsupported and requires using deprecated X11 instead of Wayland). smileBeda provided instructions in the Discord for this. We can maybe paste those into a gist or wiki article.

AdityaGarg8 commented:
The gist is already in the wiki, you can put working as well

AdityaGarg8 commented:
Also, fix the lint errors please

netnef commented:
> The gist is already in the wiki, you can put working as well

Do you have a link to that so I can include it here?

AdityaGarg8 commented:
https://wiki.t2linux.org/#guides-and-similar

I don't think we need to put it here though. But status as working should be put.

netnef commented:
The reason I think we should reference it here is because of the note we have added to it:

> Works, but isn't as great as on macOS

On the other wiki page you linked, it says that you can obtain this:

> Implement macOS-like Keyboard and trackpad experience

I don't think adding a casual mention to it would hurt. Something like this:

> Works, but isn't as great as on macOS (no force touch or palm rejection). Some models have deadzones on the edges of their trackpads where swipes along the trackpad that start in these deadzones will not be registered. Using Synaptics + X11 as per [this guide](https://wiki.t2linux.org/#guides-and-similar) could make it feel more like macOS, but comes at the cost of using a legacy display server (X11) instead of Wayland, potentially exposing other issues, like blurry display scaling.

Let me know what you think.

AdityaGarg8 commented:
I leave this to other ma

> The reason I think we should reference it here is because of the note we have added to it:
> 
> > Works, but isn't as great as on macOS
> 
> On the other wiki page you linked, it says that you can obtain this:
> 
> > Implement macOS-like Keyboard and trackpad experience
> 
> I don't think adding a casual mention to it would hurt. Something like this:
> 
> > Works, but isn't as great as on macOS (no force touch or palm rejection). Some models have deadzones on the edges of their trackpads where swipes along the trackpad that start in these deadzones will not be registered. Using Synaptics + X11 as per [this guide](https://wiki.t2linux.org/#guides-and-similar) could make it feel more like macOS, but comes at the cost of using a legacy display server (X11) instead of Wayland, potentially exposing other issues, like blurry display scaling.
> 
> Let me know what you think.

I leave this to other maintainers @Redecorating @sharpenedblade

sharpenedblade commented:
> https://gist.github.com/smileBeda/f0452f0d7f1f6d8aa772603411f7876f

This config doesnt seem to be doing anything that cant be done with libinput, we should try to get it working on wayland before puting this on the wiki (once its on the wiki people are going to use it and we "have" to support them). [touchpad preasure](https://wayland.freedesktop.org/libinput/doc/latest/touchpad-pressure-debugging.html) and [palm rejection](https://wayland.freedesktop.org/libinput/doc/latest/palm-detection.html) is supported, we just need to set the quirks, the touchpad edge bug is fixed with [this](https://github.com/t2linux/linux-t2-patches/commit/af351299c1bea3212fa934bb3eecef0c9a9e52a0) commit.

We can merge this now and change it later, or wait for proper libinput support. @Redecorating

AdityaGarg8 commented on docs/state.md:
```diff
@@ -1,35 +1,29 @@
+|Suspend|🟡 Partially working|🟢 Yes|A firmware upgrade attached to macOS Sonoma broke suspend. Some users were having difficulty with it even before Sonoma. The issue has something to do with the apple-bce driver, because suspend works when it's disabled.|[#53](https://github.com/t2linux/T2-Ubuntu-Kernel/issues/53)|
+|Audio|🟡 Partially working|🔴 No|With proper configuration audio can work, however it is not stable in some older kernels and switching between speakers and when using the microphone. Microphone volume is low in some Macs.|[apple-bce](https://github.com/t2linux/apple-bce-drv)|
+|Hybrid Graphics|🟡 Partially working|🟡 Partial|Toggling dGPU power doesn't work.|[Hybrid Graphics](https://wiki.t2linux.org/guides/hybrid-graphics/)|
+|AMD GPUs|🟡 Partially working||Changing resolution, using DRI_PRIME and doing various other things can cause crashes, but `echo high \| sudo tee /sys/bus/pci/drivers/amdgpu/0000:??:??.?/power_dpm_force_performance_level` or adding `amdgpu.dpm=0` to the kernel commandline stops these crashes.||
```
Is this still an issue?

sharpenedblade replied:
yes

AdityaGarg8 commented on docs/contribute.md:
```diff
@@ -13,5 +17,5 @@ Here is the list and links to donate some of our maintainers.
 - [NoaHimesaka1873](https://github.com/NoaHimesaka1873) - Currently maintains boot images for Arch, EndeavourOS, and Manjaro. Also maintains kernel for Manjaro too. Donate via [GitHub Sponsors](https://github.com/sponsors/NoaHimesaka1873) or [Toss (South Korea residents only)](https://toss.me/yuifmirror).
 - [Redecorating](https://github.com/Redecorating) - Currently maintains Arch kernel package, has written T2 related kernel patches and worked on T2 related kernel regressions. Donate via [PayPal](https://paypal.me/redecorating).
-- [AdityaGarg8](https://github.com/AdityaGarg8) - Current maintainer of Ubuntu. Has also helped in writing T2 kernel patches and upstreaming various patches. Donate via **Amazon Gift Cards India** ([Link for Indians](https://www.amazon.in/s?k=amazon+e-gift+voucher&adgrpid=60620499282&ext_vrnc=hi&gclid=Cj0KCQjw8O-VBhCpARIsACMvVLMW9KEhsWO9xHEvOh7VLo3lCimvvJJqaKdqwj-OlN_ZAMp3ZIztQ0AaAtGAEALw_wcB&hva ...
+- [AdityaGarg8](https://github.com/AdityaGarg8) - Current maintainer of Ubuntu. Has also Current maintainer of Ubuntu. Has also helped in writing T2 kernel patches and upstreaming various patches. Donate via **Amazon Gift Cards India** ([Link for Indians](https://www.amazon.in/s?k=amazon+e-gift+voucher&adgrpid=60620499282&ext_vrnc=hi&gclid=Cj0KCQjw8O-VBhCpARIsACMvVLMW9KEhsWO9xHEvOh7VLo3lCimvvJJqa ...
```
I messed up a line here, could you please correct it

AdityaGarg8 reviewed (commented):
Let's merge this after addressing all this?

netnef commented:
> > https://gist.github.com/smileBeda/f0452f0d7f1f6d8aa772603411f7876f
> 
> This config doesnt seem to be doing anything that cant be done with libinput, we should try to get it working on wayland before puting this on the wiki (once its on the wiki people are going to use it and we "have" to support them). [touchpad preasure](https://wayland.freedesktop.org/libinput/doc/latest/touchpad-pressure-debugging.html) and [palm rejection](https://wayland.freedesktop.org/libinput/doc/latest/palm-detection.html) is supported, we just need to set the quirks, the touchpad edge bug is fixed with [this](https://github.com/t2linux/linux-t2-patches/commit/af351299c1bea3212fa934bb3eecef0c9a9e52a0) commit.
> 
> We can merge this now and change it later, or wait for proper libinput support. @Redecorating

The config is already on the wiki: https://wiki.t2linux.org/#guides-and-similar

Perhaps we should clarify that it's unsupported by the t2linux maintainers there.

netnef commented:
I updated two pages to address concerns from @sharpenedblade. Let me know what you think.

* https://netnef.github.io/wiki/#guides-and-similar
* https://netnef.github.io/wiki/state/

sharpenedblade commented:
The deadzone problem is fixed a few days ago, we can merge once thats removed

netnef commented:
I removed the deadzones remark. @AdityaGarg8 are we good to merge?

AdityaGarg8 commented:
Looks like we are

netnef commented:
👍 Thank you all for the help. Glad we got these pages updated. Long overdue.
